src/main/java/org/sagebionetworks/bridge/services/Exporter3Service.java

```diff
@@ -968,66 +971,60 @@ private void exportUpload(String appId, String recordId) {
-//        printWriter.close();
-        PrintWriter pw = null;
-        File file = File.createTempFile("timelinefor" + studyId, ".txt");
+        File outputFile = File.createTempFile("timelineFor" + studyId, ".txt");
+        CharSink charSink = Files.asCharSink(outputFile, Charsets.UTF_8, FileWriteMode.APPEND);
         try {
-            FileOutputStream fos = new FileOutputStream(file);
-            pw = new PrintWriter(fos);
-            pw.print(node.toString());
-            pw.close();
-            pw = null;
-        } finally {
-            if (pw != null) {
-                pw.close();
-            }
+            charSink.write(node.toString());
+        } catch (IOException e) {
+            e.printStackTrace();
         }
-        CloudProviderFileHandleInterface markdown = synapseClient.multipartUpload(file,
-                null, false, false);
-        System.out.println(markdown);
-
-//        markdown.setStorageLocationId(exporter3Config.getStorageLocationId());
-//        markdown.setBucketName(rawHealthDataBucket);
-//        markdown.setKey(appId + '/' + studyId + '/' + "timeline");
-//        byte[] md5 = Base64.getUrlDecoder().decode(node.toString());
-//        String hexMd5 = Hex.encodeHexString(md5);
-//        markdown.setContentMd5(hexMd5);
-//        markdown = synapseHelper.createS3FileHandleWithRetry(markdown);
-
-        // If first time exporting the timeline for the study:
+        CloudProviderFileHandleInterface markdown = synapseClient.multipartUpload(outputFile,
+                exporter3Config.getStorageLocationId(), false, false);
+
+        // If first time exporting the timeline for the study, create a new wiki page in Synapse;
+        // If wiki page already exists, update the existing wiki page.
         if (exporter3Config.getWikiPageId() == null) {
             V2WikiPage wiki = new V2WikiPage();
-//            wiki.setId("wikiFor" + studyId);
             wiki.setTitle("Exported Timeline for " + studyId);
-            exporter3Config.setWikiPageId(wiki.getId());
             wiki.setMarkdownFileHandleId(markdown.getId());
-
             wiki = synapseClient.createV2WikiPage(exporter3Config.getProjectId(), ObjectType.ENTITY, wiki);
+            exporter3Config.setWikiPageId(wiki.getId());
+            isConfigModified = true;
         } else {
-            // If the wiki page for the timeline to export already exists:
             WikiPageKey key = WikiPageKeyHelper.createWikiPageKey(exporter3Config.getProjectId(), ObjectType.ENTITY, exporter3Config.getWikiPageId());
             V2WikiPage getWiki = synapseClient.getV2WikiPage(key);
             getWiki.setMarkdownFileHandleId(markdown.getId());
             synapseClient.updateV2WikiPage(exporter3Config.getProjectId(), ObjectType.ENTITY, getWiki);
         }
+
+        // If exporter3 config is modified, update study.
+        if (isConfigModified) {
+            studyService.updateStudy(appId, study);
+        }
```

Comment:
If you move the call to studyService.updateStudy() to inside the if block for if (exporter3Config.getWikiPageId() == null), then you don't need the isConfigModified flag.